archive/0000-staking-contract.md

```diff
@@ -0,0 +1,176 @@
+- Proposal Name: staking-contract
+- Start Date: 2020-01-12
+- NEP PR: [nearprotocol/neps#0000](https://github.com/nearprotocol/neps/pull/0000)
+- Issue(s): link to relevant issues in relevant repos (not required).
+
+# Summary
+[summary]: #summary
+
+Provides standard and reference internal design for staking contract with delegation.
+
+# Motivation
+[motivation]: #motivation
+
+In NEAR contracts and accounts are the same object. This allows to have a contract that stakes NEAR with it's balance.
+This is useful to implement delegation, custom reward dispersion, liquid staking and various derivatives on stake and more.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The simplest example of staking contract would be a delegation contract.
+
+Let's define actors:
+- The staking pool contract account `staking-pool`. A key-less account with the contract that pools funds.
+- The owner of the staking contract `owner`. Owner runs the validator node on behalf of the staking pool account.
+- A delegator `user1`. The account who wants to stake their fund to the pool.
+
+The owner can setup such contract and validate on behalf of this contract in their node.
+Any other user can send their tokens to the contract, which will be pooled together and increase the total stake.
+These users would accrue rewards (subtracted fees set by the owner).
+Then they can unstake and withdraw their balance after some unlocking period.
```

Comment:
```suggestion
Usually, owner owns the contract and is the validator. Delegators will stake through the contract. The stake will be pooled and if enough stake is collected, it will allow the validator to be an active validator and consequently receive staking rewards. The default implementation is fully permission-less and automates the staking reward distribution as well as the validator fees.
Delegators can unstake and withdraw their balance after the unlocking period.
```